### ISSUE TITLE
How does "insert `mod ...`" decide where to insert the mod?

### ISSUE THREAD

**Issue by idanarye** [rust-lang/rust-analyzer#14469]
In past projects, rust-analyzer's `insert mod` code action was inserting the `mod ...;` in `lib.rs`. I've started a new project today, and for some reason it now inserts them at `main.rs` instead.

How does rust-analyzer decide where to insert it? What do I need to do to make it insert them at `lib.rs`?

**Comment by Veykril** [comment]
crate roots are checked here, looks like we just pick the first we find nowadays instead of trying to prefer lib.rs over main.rs https://github.com/rust-lang/rust-analyzer/blob/236576227a299fd19ba836b1834ab50c948af994/crates/ide-diagnostics/src/handlers/unlinked_file.rs#L59-L90

**Comment by idanarye** [comment]
So, this is just luck? For some reason this time it read them in different order?

Is there any way I can override this?

**Comment by Veykril** [comment]
no, this code was reworked a few months ago. We should prioritize `lib.rs` over `main.rs` but that priority got lost when this was rewritten.
